samples/matter/lock/sample.yaml

```diff
@@ -4,33 +4,91 @@ sample:
 tests:
   sample.matter.lock.debug:
     build_only: true
+    extra_args: CONFIG_CHIP_LIB_SHELL=y
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
-    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
     tags: ci_build
-  sample.matter.lock.no_dfu:
+  sample.matter.lock.release:
     build_only: true
-    extra_args: CONF_FILE=prj_no_dfu.conf
+    extra_args: CONF_FILE=prj_release.conf
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
-      - nrf21540dk_nrf52840
     platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
     tags: ci_build
-  sample.matter.lock.release:
+
+  sample.matter.lock.debug.ffs:
     build_only: true
-    extra_args: CONF_FILE=prj_release.conf
+    extra_args: CONFIG_CHIP_LIB_SHELL=y CONFIG_CHIP_COMMISSIONABLE_DEVICE_TYPE=y CONFIG_CHIP_ROTATING_DEVICE_ID=y CONFIG_CHIP_DEVICE_TYPE=10
+    integration_platforms:
+      - nrf52840dk_nrf52840
+      - nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
+    tags: ci_build
+  sample.matter.lock.release.ffs:
+    build_only: true
+    extra_args: CONF_FILE=prj_release.conf CONFIG_CHIP_COMMISSIONABLE_DEVICE_TYPE=y CONFIG_CHIP_ROTATING_DEVICE_ID=y CONFIG_CHIP_DEVICE_TYPE=10
+    integration_platforms:
+      - nrf52840dk_nrf52840
+      - nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
+    tags: ci_build
+
+  sample.matter.lock.debug.smp_dfu:
+    build_only: true
+    extra_args: CONFIG_CHIP_LIB_SHELL=y CONFIG_CHIP_DFU_OVER_BT_SMP=y
+    integration_platforms:
+      - nrf52840dk_nrf52840
+      - nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
+    tags: ci_build
+  sample.matter.lock.release.smp_dfu:
+    build_only: true
+    extra_args: CONF_FILE=prj_release.conf CONFIG_CHIP_DFU_OVER_BT_SMP=y
     integration_platforms:
```

Comment:
IMHO it's not worth building all variants (debug + SMP DFU and release + SMP DFU) in the integration plan. Building only debug variants + a single release version should be enough to verify the build.

Reply:
I thin we're going to handle this another time, this is only unification of sample.yamls, and NCS.json in TAS. These builds are sum of builds in TAS and here.